### ISSUE TITLE
how to apply the test_epoch and output the classification map?

### ISSUE THREAD

**Issue by AlanMorningLight** [danfenghong/IEEE_TGRS_SpectralFormer#1]
Dear danfeng,
Thanks a lot for your sharing codes.  I have a few questions：① Why is the sample set divided into only two parts: training set and test set? Why is there no validation set? ② How to use the test_epoch function? ③ How to output the entire classification map? Can you provide the code? Thank you very much for your reply.

**Comment by danfenghong** [comment]
Hi,
Thank you very much for your interest in our work!
Regarding these issues,
1. You can seperate the validation set for sure, if you would like to tune
the parameters using the validation set.
2. I did not get your point? is there a difficulty to use this function?
3. for the whole classification map, you only input all samples into the
trained network and then use the matlab for showing.

Thanks again!
Cheers,
Danfeng

Wenchao QI ***@***.***> 于2021年11月24日周三 上午11:22写道：

**Comment by AlanMorningLight** [comment]
Hi, 
With your help, I have successfully run the program. But I have another question: 

- Both training samples and test samples should contain 16 categories of Indian Pines dataset, but when using ''train_and_test_label'' function to generate the label corresponding to the training and test samples, why is the label value ''0-15'' instead of ''1-16''? 
- I think, '' y_train.append(i)'' and '' y_test.append(i)'' should be '' y_train.append(i+1)'' and '' y_test.append(i+1)'', respectively. 
- And y_true contains the background value 0, a total of 17 categories, and the label value is 0-16, so y_true. append (i) is correct in the demo.py.

Thank you for your reply!
best regards,
Wenchao

**Comment by danfenghong** [comment]
Hi,

I think this is basic grammar for python to how to define such a vector.
You can simply check some examples to how to define this label information.
Cheers,
Danfeng

Wenchao QI ***@***.***> 于2021年11月27日周六 下午11:09写道：
